connectors/src/connectors/notion/lib/parents.ts

```diff
@@ -116,39 +125,33 @@ export async function updateAllParentsFields(
  * updated
  */
 async function getPagesToUpdate(
-  pageOrDbs: (NotionPage | NotionDatabase)[],
+  createdOrMovedNotionPageIds: string[],
+  createdOrMovedNotionDatabaseIds: string[],
   dataSourceConfig: DataSourceConfig
-): Promise<NotionPage[]> {
-  const pagesToUpdate: NotionPage[] = [];
+): Promise<Set<string>> {
+  const pageIdsToUpdate: Set<string> = new Set([
+    ...createdOrMovedNotionPageIds,
+  ]);
```

Comment:
not sure the change of logic from while to for works here
Are you sure that all descendants are updated? not just children?

Reply:
why did you need to change this part of parents?

Reply:
There's still a loop below, as far as I can tell, this + the loop below is equivalent ?

Reply:
Ok I think I see it my bad